Near-final updates (#31)
ivoa/HighEnergyObsCoreExt: HighEnergyObsCoreExt.tex

The goal is to explore elements needed to reliably discover and select \gls{HEA} data through \gls{IVOA} interfaces. This requires defining an extension to ObsCore with the possibility to use the DataLink mechanism and to enhance vocabularies of keywords for ObsCore and DataLink. We suggest that, if an attribute is unique to \gls{HEA} data, that element should appear in an \gls{HEA} ObsCore extension. Whereas, if an attribute makes sense for more than one domain and can be shared across those domains, then that element should be added to the base ObsCore model. This note proposes recommendations in both of these categories. We also discuss enhancements to the vocabulary of data\/ products, DataLink semantics, Unified Content Descriptors (UCDs), and MIME-types to correctly represent \gls{HEA} data. Topics related to the Registry are currently outlined in the proposed Radio extension document and are not discussed here.

\section{High Energy Astrophysics Data}

> \gls{HEA} data include observations obtained using photon detectors covering X-ray (from $\sim$0.1 keV to $\sim$120 keV) through gamma-ray (from 120 keV up to $\gtrsim$ PeV) energies, as well as cosmic-ray and astrophysical neutrino ($\gtrsim$ TeV) detectors, or other messengers related to \gls{HEA} phenomena. The domain is now sufficiently mature to provide open data that are science-ready and work with open analysis tools ({\em e.g.\/}, CIAO \citep{2006SPIE.6270E..1VF} or Gammapy \citep{gammapy:2023}). The science output of the \gls{HEA} domain already includes high-level products such as images, cubes, spectra, and time series such as light curves and time-resolved spectra. Additional data products include fitted sky models with a spatial, spectral, and/or temporal component(s), along with their confidence intervals or confidence limits, and covariance matrices.   Finally, multiple \gls{HEA} instruments produce source catalogs and surveys covering up to the full the sky, which include maps of photon or particle flux, exposure, sensitivity, and aperture-photometry likelihood profiles.

Reviewer: ```suggestion
\gls{HEA} data include observations obtained using photon detectors covering X-ray (from $\sim$0.1 keV to $\sim$120 keV) through gamma-ray (from 120 keV up to $\gtrsim$ PeV) energies, as well as cosmic-ray and astrophysical neutrino ($\gtrsim$ TeV) detectors, or other messengers related to \gls{HEA} phenomena. The domain is now sufficiently mature to provide open data that are science-ready and work with open analysis tools ({\em e.g.\/}, CIAO \citep{2006SPIE.6270E..1VF} or Gammapy \citep{gammapy:2023}). The science output of the \gls{HEA} domain already includes advanced products such as images, cubes, spectra, and time series such as light curves and time-resolved spectra. Additional data products include fitted sky models with a spatial, spectral, and/or temporal component(s), along with their confidence intervals or confidence limits, and covariance matrices.   Finally, multiple \gls{HEA} instruments produce source catalogs and surveys covering up to the full the sky, which include maps of photon or particle flux, exposure, sensitivity, and aperture-photometry likelihood profiles.
```

\end{quote}

We note that the term {\bf event} has caused confusion in the past, since ``event'' may also used to describe notifications ({\em e.g.\/}, as in ``VOEvent'') of astrophysical events such as supernova explosions.  Such ``events'' are quite different from the particle detection events being described herein. Using {\bf event-list} will help to resolve this ambiguity.

%An {\bf event-bundle} might for example consist of an {\bf event-list} and the associated {\bf response-functions} (see below) used to calibrate the dataset; alternatively an {\bf event-bundle} may include the {\bf event-list} and associated  data products necessary for the user to create the {\bf response-functions} (for those X-ray cases where detailed knowledge of the scientific use case — for example, the user’s selection of events — may be required to compute the responses).\\
%particle-detection

> In addition to {\em dataproduct\_type\/} terms that focus on event data, we note that existing ObsCore definitions do not adequately span the breadth of ``advanced data products'' (typically with {\em calib\_level\/} $\ge$ 3) that may be generated from astronomical observations by users or observatories. The computational complexity of analyzing \gls{HEA} data robustly in the extreme Poisson regime ({\em e.g.\/}, Bayesian X-ray aperture photometry applied simultaneously to multiple overlapping detections and observations) means that data providers may choose to provide such analysis products directly to the end user. For example, the Chandra Source Catalog includes 38 types of advanced data products (for a total of $\sim\!90$ million files) and $\sim\!50$\% of these data product types are not well represented by a  {\em dataproduct\_type\/} value that allows for meaningful data discovery. Users will certainly want to discover these data products independently from the associated progenitor observation data (and many of these data products combine data from multiple observations). We therefore propose the following additional {\em dataproduct\_type\/} (or {\em dataproduct\_subtype\/}) terms for these advanced data products, and note that these terms will certainly be useful independent of waveband ({\em i.e.\/}, they can be equally applicable to UV/optical, IR, and radio datasets):

Reviewer: ```suggestion
In addition to {\em dataproduct\_type\/} terms that focus on event data, we note that existing ObsCore definitions do not adequately span the breadth of ``advanced data products'' (typically with {\em calib\_level\/} $\ge$ 3) that may be generated from astronomical observations by users or observatories. The computational complexity of analyzing \gls{HEA} data robustly in the extreme Poisson regime ({\em e.g.\/}, Bayesian X-ray aperture photometry applied simultaneously to multiple overlapping detections and observations, Frequentist adjustment of a model of electron populations on multi-wavelength data spanning from X-rays to PeV gamma rays) means that data providers may choose to provide such analysis products directly to the end user. For example, the Chandra Source Catalog includes 38 types of advanced data products (for a total of $\sim\!90$ million files) and $\sim\!50$\% of these data product types are not well represented by a  {\em dataproduct\_type\/} value that allows for meaningful data discovery. Users will certainly want to discover these data products independently from the associated progenitor observation data (and many of these data products combine data from multiple observations). We therefore propose the following additional {\em dataproduct\_type\/} (or {\em dataproduct\_subtype\/}) terms for these advanced data products, and note that these terms will certainly be useful independent of waveband ({\em i.e.\/}, they can be equally applicable to UV/optical, IR, and radio datasets):
```

Reviewer: I have added a VHE example, such that anyone can be convinced that the need of these products is from the whole HEA group.

{\bf region}: a dataset that includes an encoding of (one or more) regions of parameter space, for example a spatial region or a region of phase space covered by a dataset. The set of dimensions represented by the region can be arbitrary.

{\bf response-function}: a dataset that records a mapping from a physical quantity to an observable quantity. For \gls{HEA}, this may be the components of the composite \gls{IRF} such as an Auxiliary Response File ({\bf arf}), Redistribution Matrix File ({\bf rmf}), Effective Area ({\bf aeff}), Energy Dispersion ({\bf edisp}), the Background Rate ({\bf bkgrate}). The Point Spread Function ({\bf psf}) is a response function that is generally applicable across multiple wavebands. While these datasets may generally be represented as an N-dimensional data cube, designating them as {\bf response-functions} enhances data discovery for very common types of \gls{HEA} dataset (see the use cases in Appendix~\ref{sec:uc}).

\end{quote}

> The {\bf measurements} {\em dataproduct\_type\/} is quite useful for many different types of advanced data products (that may be derived from multiple observations). But users of those products often may not be interested the progenitor datasets, especially if many advanced data products are extracted from a single or a few progenitors ({\em e.g.\/}, measurements associated with sources detected in a single observation field). We propose to delete the caveat associated with {\em dataproduct\_type\/} = ``measurements'' in the ObsCore IVOA Recommendation (\S~4.1.1) that requires the derived data products be exposed ``{\bf together} with the progenitor observation dataset''.

Reviewer: ```suggestion
The {\bf measurements} {\em dataproduct\_type\/} is quite useful for many different types of advanced data products (that may be derived from multiple observations). But users of those products often may not be interested the progenitor datasets, especially if many advanced data products are extracted from a single or a few progenitors ({\em e.g.\/}, measurements associated with sources detected in a single observation field). We propose to delete the caveat associated with {\em dataproduct\_type\/} = ``measurements'' in the ObsCore IVOA Recommendation (\S~4.1.1) that requires the derived data products be exposed ``{\bf together} with the progenitor observation dataset''. The recovery of the progenitor observation dataset(s) can be achieved with the Provenance information.
```

\subsection{{\em dataproduct\_subtype}}

> The optional attribute {\em dataproduct\_subtype} may be used by the data provider to specify more precisely the scientific nature of a data product.  Although no vocabulary is defined for {\em dataproduct\_subtype\/}, we recommend that data providers formulate and use a standardized vocabulary for this attribute for data products that are commonly used in \gls{HEA}\null.  We have proposed several terms in \S~5 for commonly used \gls{HEA} {\bf response-function} types ({\em e.g.\/}, {\bf aeff}, {\bf edisp}, {\bf psf}), but additional terms could be standardized for other common data products.  For example, standardizing using {\bf exposuremap} for an exposure map would enable queries such as  ({\em dataproduct\_type\/} = {\bf image}) AND ({\em dataproduct\_subtype\/} = {\bf exposuremap}) to work across multiple facilities.  Other possible terms could include {\bf significancemap} for a significance map, and {\bf probabilitymap} for aprobability map.

Reviewer: ```suggestion
The optional attribute {\em dataproduct\_subtype} may be used by the data provider to specify more precisely the scientific nature of a data product.  Although no vocabulary is defined for {\em dataproduct\_subtype\/}, we recommend that data providers formulate and use a standardized vocabulary for this attribute for data products that are commonly used in \gls{HEA}\null.  We have proposed several terms in \S~5 for commonly used \gls{HEA} {\bf response-function} types ({\em e.g.\/}, {\bf aeff}, {\bf edisp}, {\bf psf}, {\bf bkgrate}), but additional terms could be standardized for other common data products.  For example, standardizing using {\bf exposuremap} for an exposure map would enable queries such as  ({\em dataproduct\_type\/} = {\bf image}) AND ({\em dataproduct\_subtype\/} = {\bf exposuremap}) to work across multiple facilities.  Other possible terms could include {\bf significancemap} for a significance map, {\bf probabilitymap} for a probability map, and {\bf exclusionmap} for the exclusion maps used to adjust TeV background models.
```

In the example {\em o\_ucd\/} above, the UCD {\em instr.event.pulseHeight\/} is used to represent the detector Pulse Height Amplitude (PHA)\null. There is currently no UCD defined for a raw measure like PHA, but we propose the addition of {\em instr.event.pulseHeight\/} to the UCDs list vocabulary, together with other UCDs that are relevant for \gls{HEA} data, in \S~\ref{sec:UCDs}.  Several additional UCDs, including electromagnetic spectrum, physical quantities, and statistical parameters UCDs, are also proposed in \S~\ref{sec:UCDs} that are relevant for \gls{HEA} data products but could also be of use for other domains such as cosmology.

> Advanced data products may similarly record multiple observables that can only be differentiated through their UCDs. For example, a Chandra Source Catalog {\bf pdf} dataset for a detection may include multiple marginalized probability density functions computed using a Bayesian X-ray aperture photometry algorithm in units of net counts, net count rates, photon fluxes, and energy fluxes in multiple apertures.  The observables recorded in the different MPDFs may be distinguished by their UCDs which then become relevant for data discovery when a user is searching for specific aperture photometry datasets.

Reviewer: ```suggestion
Advanced data products may similarly record multiple observables that can only be differentiated through their UCDs. For example, a Chandra Source Catalog {\bf pdf} dataset for a detection may include multiple Marginalized Probability Density Functions (MPDFs) computed using a Bayesian X-ray aperture photometry algorithm in units of net counts, net count rates, photon fluxes, and energy fluxes in multiple apertures.  The observables recorded in the different MPDFs may be distinguished by their UCDs which then become relevant for data discovery when a user is searching for specific aperture photometry datasets.
```

\subsection{{\em obs\_mode}}

Many \gls{HEA} instruments may be configured using multiple observing modes and these observing modes may significantly impact the structure and characteristics ({\em e.g.\/}, calibration accuracy) of the resulting observation datasets. For example, the Chandra ACIS instrument typically produces {\bf event-list}s with 2-dimensional spatial coordinates ({\em i.e.\/}, imaging) but has an observation mode that continuously reads-out the detector, effectively producing an {\bf event-list} with a single spatial dimension (the other spatial dimension is collapsed); users looking only for imaging data may want to restrict their queries to exclude the latter observing mode.

We propose to add an optional attribute {\em obs\_mode\/} that allows the data provider to specify the observation mode for an observation.  Constraints on observation mode can provide a simple way to discover data sets for a specific facility/instrument combination.  We note that permissible {\em obs\_mode\/} values will vary from facility to facility and from instrument to instrument.

\subsection{{\em tracking\_type}}

> Since most \gls{HEA} facilities record the arrival times of particle events, data accumulated when the telescope pointing moves or rotates relative to the sky during an observation can be located on the sky.  The way in which the telescope moves may impact how the data should be processed, especially for observations in which multiple different astrophysical sources may be present within the field of view ({\em e.g.\/}, a solar system target moving relative to the fixed celestial background).  {\em Tracking\/} is defined by the normal motion of the optical axis of the telescope during an observation: fixed to a celestial coordinate (the most common case for ground telescopes with movable mounts, or those in space),  fixed to a horizontal coordinate (also called a ``drift scan'', and the standard case for telescopes without movable mounts),  or fixed to a solar system body ({\em e.g.\/}, the moon, a planet, or a comet) position.

Reviewer: ```suggestion
Since most \gls{HEA} facilities record the arrival times of particle events, data accumulated when the telescope pointing moves or rotates relative to the sky during an observation can be located on the sky.  The way in which the telescope moves may impact how the data should be processed, especially for observations in which multiple different astrophysical sources may be present within the field of view ({\em e.g.\/}, a solar system target moving relative to the fixed celestial background).  {\em Tracking\/} is defined by the normal motion of the optical axis of the telescope or the field of view center of a facility during an observation: fixed to a celestial coordinate (the most common case for ground telescopes with movable mounts, or those in space),  fixed to a horizontal coordinate (also called a ``drift scan'', and the standard case for telescopes without movable mounts),  or fixed to a solar system body ({\em e.g.\/}, the moon, a planet, or a comet) position.
```

\subsection{{\em scan\_mode}}

Some \gls{HEA} facilities can obtain observations using different spatial scan modes that will affect the content of the observation, leading to different instrument responses and special analysis schemes.  We propose to add an optional attribute {\em scan\_mode\/} to distinguish these modes.  Constraints on {\bf scan\_mode} can provide a simple way to discover data sets for a specific facility/instrument combination.  We propose predefined {\em scan\_mode\/} values for \gls{HEA} data include the following:
\begin{itemize}
	\item \texttt{on-source}: pointed observation;

> 	\item \texttt{on-off}: switched observations between two spatial positions (source and background);

Reviewer: ```suggestion
	\item \texttt{on-off}: switched observations between on-source spatial position and off-source position(s);
```

The shape of any statistical distribution is an essential quantity for interpreting the meaning of any statistical properties.  Too often a Gaussian distribution or a distribution that can be simply characterized by a simple set of moments ({\em e.g.\/}, mean, variance, skewness, kurtosis) are assumed, but in the extreme Poisson regime common in \gls{HEA} these assumptions are often invalid.  We propose adding a UCD {\em stat.distribution\/} to identify a quantity that defines the distribution of a statistical variable such as  a likelihood profile.

\subsubsection{Evolution of UCD list}

The proposed new UCD entries are listed in Table~\ref{tab:he_ucds} with their descriptions, while proposed revisions to existing UCD entries are  listed in Table~\ref{tab:upgrade_he_ucds}.

\begin{longtable}{p{0.05\linewidth}p{0.35\linewidth}p{0.6\linewidth}}
\sptablerule
\textbf{}  &  \textbf{UCD word} & \textbf{Description}\cr
\sptablerule
S & {\em em.gamma.he\/}   & High-Energy gamma ray (100 MeV -- 10 GeV)  \cr
S & {\em em.gamma.vhe\/}  & Very-High-Energy gamma ray (10 GeV -- 100 TeV)  \cr

> S & {\em em.gamma.uhe\/}  & Ultra-High-Energy gamma ray ($>100$ TeV)  \cr

Reviewer: ```suggestion
S & {\em em.gamma.uhe\/}  & Ultra-High-Energy gamma ray ($>100$ TeV)  \cr
S & phys.particle.he   &High-Energy particles (100 MeV -- 10 GeV)  \cr
S & phys.particle.vhe  &Very-High-Energy particles (10 GeV -- 100 TeV)  \cr
S & phys.particle.uhe  &Ultra-High-Energy particles ($>100$ TeV) \cr
```

Q & {\em instr.event\/} & Particle event detection \cr
Q & {\em instr.event.grade\/} & Particle event grade \cr
Q & {\em instr.pulseHeight\/} & Pulse height amplitude measure  \cr                                   
Q & {\em instr.event.type\/} & Particle event type \cr                                                    

> E & {\em phot.count.density\/} & Count or particle flux density (dimensionality: $\rm [L^{-2}\,T^{-1}\,E^{-1}]$)  \cr

Reviewer: ```suggestion
E & {\em phot.count.density\/} & Count flux density or differential flux (dimensionality: $\rm [L^{-2}\,T^{-1}\,E^{-1}]$)  \cr
```

E & {\em phot.count.density.sb\/} & Count or particle flux density surface brightness (dimensionality: $\rm [L^{-2}\,T^{-1}\,E^{-1}\,\hbox{sr}^{-1}]$)  \cr
E & {\em phot.count.radiance\/} & Count or particle flux radiance (dimensionality: $\rm [L^{-2}\,T^{-1}\,\hbox{sr}^{-1}]$)  \cr
E & {\em phot.count.sb\/} & Count or particle flux surface brightness (dimensionality: $\rm [L^{-2}\,T^{-1}\,\hbox{sr}^{-1}]$)  \cr
E & {\em phot.flux.energy\/} & Energy flux or irradiance (dimensionality: $\rm [M\,T^{-3}]$)  \cr
E & {\em phot.flux.energy.density\/} & Energy flux density (dimensionality: $\rm [M\,T^{-3}\,E^{-1}]$)  \cr

> E & {\em phot.flux.energy.density.sb\/} & Energy flux density surface brightness (dimensionality: $\rm [M\,T^{-3}\,E^{-1}\,\hbox{sr}^{-1}]$)  \cr                                                                 

Reviewer: ```suggestion
E & {\em phot.energy.density.sb\/} & Energy flux density surface brightness (dimensionality: $\rm [L^{-2}\,T^{-1}\,E^{-1}\,\hbox{sr}^{-1}]$)  \cr                                                                 
```
